// mc-components/dist/components/VideoPlayerScreen/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactTransitionGroup = require("react-transition-group");

var _classnames = _interopRequireDefault(require("classnames"));

var _VideoPlayerPortalScreen = _interopRequireDefault(require("../VideoPlayerPortalScreen"));

var _constants = require("../constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var VideoPlayerScreen = function VideoPlayerScreen(_ref) {
  var isActive = _ref.isActive,
      variation = _ref.variation,
      children = _ref.children,
      videoRoot = _ref.videoRoot;

  if (!videoRoot) {
    return null;
  }

  var classNames = (0, _classnames["default"])('bc-player__screen', _defineProperty({}, "bc-player__screen--".concat(variation), variation));
  return _react["default"].createElement(_VideoPlayerPortalScreen["default"], {
    videoRoot: videoRoot
  }, _react["default"].createElement(_reactTransitionGroup.CSSTransition, {
    appear: isActive,
    "in": isActive,
    classNames: "bc-player__screen-",
    timeout: 500
  }, _react["default"].createElement("div", {
    className: classNames
  }, children)));
};

VideoPlayerScreen.propTypes = {
  children: _constants.PROP_TYPE_CHILDREN.isRequired,
  variation: _propTypes["default"].oneOf(['endscreen', 'beforescreen', 'pausescreen']),
  videoRoot: _propTypes["default"].object,
  isActive: _propTypes["default"].bool
};
VideoPlayerScreen.defaultProps = {
  isActive: false
};
var _default = VideoPlayerScreen;
exports["default"] = _default;
