// mc-components/dist/components/Slider/Slider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Slider = function Slider(_ref) {
  var value = _ref.value,
      buffer = _ref.buffer,
      onChange = _ref.onChange;

  var handleScrubberInteraction = function handleScrubberInteraction(scrubber) {
    return function (event) {
      var _scrubber$getBounding = scrubber.getBoundingClientRect(),
          scrubberLeft = _scrubber$getBounding.left,
          scrubberWidth = _scrubber$getBounding.width;

      var eventLeft = event.clientX;
      var left = Math.min(Math.max(0, eventLeft - scrubberLeft), scrubberWidth);
      var value = left / scrubberWidth;
      onChange(value);
    };
  };

  var handleScrubberMouseDown = function handleScrubberMouseDown(event) {
    var listener = handleScrubberInteraction(event.currentTarget);
    handleScrubberInteraction(event.currentTarget)(event);
    window.addEventListener('mousemove', listener);
    window.addEventListener('mouseup', function () {
      window.removeEventListener('mousemove', listener);
    });
  };

  var bufferPerc = "".concat(buffer * 100, "%");
  var valuePerc = "".concat(value * 100, "%");
  return _react["default"].createElement("div", {
    className: "mc-slider",
    onMouseDown: handleScrubberMouseDown
  }, _react["default"].createElement("div", {
    className: "mc-slider__bar"
  }, _react["default"].createElement("div", {
    className: "mc-slider__buffer",
    style: {
      width: bufferPerc
    }
  }), _react["default"].createElement("div", {
    className: "mc-slider__fill",
    style: {
      width: valuePerc
    }
  }), _react["default"].createElement("div", {
    className: "mc-slider__handle-container"
  }, _react["default"].createElement("div", {
    className: "mc-slider__handle",
    style: {
      left: valuePerc
    }
  }))));
};

Slider.propTypes = {
  value: _propTypes["default"].number.isRequired,
  buffer: _propTypes["default"].number,
  onChange: _propTypes["default"].func.isRequired
};
var _default = Slider;
exports["default"] = _default;
