// mc-components/dist/components/Forms/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PROP_TYPE_REDUX_FORM_ELEMENT = exports.PROP_TYPE_REDUX_FORM_META = exports.PROP_TYPE_REDUX_FORM_INPUT = exports.PROP_TYPE_OPTIONS = exports.PROP_TYPE_OPTION = exports.PROP_TYPE_INPUT = exports.STATE_SUCCESS = exports.STATE_ERROR = exports.STATE_DEFAULT = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var STATE_DEFAULT = 'default';
exports.STATE_DEFAULT = STATE_DEFAULT;
var STATE_ERROR = 'error';
exports.STATE_ERROR = STATE_ERROR;
var STATE_SUCCESS = 'success';
exports.STATE_SUCCESS = STATE_SUCCESS;

var PROP_TYPE_MESSAGE = _propTypes["default"].oneOfType([_propTypes["default"].bool, _propTypes["default"].string]);

var PROP_TYPE_INPUT = {
  disabled: _propTypes["default"].bool,
  error: PROP_TYPE_MESSAGE,
  help: PROP_TYPE_MESSAGE,
  label: _propTypes["default"].string,
  maxlength: _propTypes["default"].number,
  name: _propTypes["default"].string,
  value: _propTypes["default"].string,
  success: PROP_TYPE_MESSAGE,
  touched: _propTypes["default"].bool,
  onBlur: _propTypes["default"].func,
  onChange: _propTypes["default"].func,
  onFocus: _propTypes["default"].func
};
exports.PROP_TYPE_INPUT = PROP_TYPE_INPUT;

var PROP_TYPE_OPTION = _propTypes["default"].shape({
  label: _propTypes["default"].string.isRequired,
  value: _propTypes["default"].string.isRequired
});

exports.PROP_TYPE_OPTION = PROP_TYPE_OPTION;

var PROP_TYPE_OPTIONS = _propTypes["default"].arrayOf(PROP_TYPE_OPTION);

exports.PROP_TYPE_OPTIONS = PROP_TYPE_OPTIONS;

var PROP_TYPE_REDUX_FORM_INPUT = _propTypes["default"].shape({
  name: _propTypes["default"].string.isRequired,
  onChange: _propTypes["default"].func.isRequired,
  value: _propTypes["default"].any.isRequired
});

exports.PROP_TYPE_REDUX_FORM_INPUT = PROP_TYPE_REDUX_FORM_INPUT;

var PROP_TYPE_REDUX_FORM_META = _propTypes["default"].shape({
  error: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].arrayOf(_propTypes["default"].string)])
});

exports.PROP_TYPE_REDUX_FORM_META = PROP_TYPE_REDUX_FORM_META;
var PROP_TYPE_REDUX_FORM_ELEMENT = {
  className: _propTypes["default"].string,
  input: PROP_TYPE_REDUX_FORM_INPUT,
  meta: PROP_TYPE_REDUX_FORM_META
};
exports.PROP_TYPE_REDUX_FORM_ELEMENT = PROP_TYPE_REDUX_FORM_ELEMENT;
