// mc-components/dist/components/Legacy/Carousel/arrow.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Arrow = _interopRequireDefault(require("../../Icons/Arrow"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var CarouselArrow = function CarouselArrow(_ref) {
  var direction = _ref.direction,
      className = _ref.className,
      onClick = _ref.onClick,
      isHidden = _ref.isHidden;
  var classNames = (0, _classnames["default"])('mc-carousel-arrow', _defineProperty({}, className, Boolean(className)), _defineProperty({}, "mc-carousel-arrow--".concat(direction), Boolean(direction)));
  return _react["default"].createElement("button", {
    className: classNames,
    onClick: onClick,
    style: isHidden ? {
      opacity: 0
    } : null
  }, _react["default"].createElement(_Arrow["default"], {
    className: "mc-carousel-arrow__image"
  }));
};

CarouselArrow.propTypes = {
  className: _propTypes["default"].string,
  onClick: _propTypes["default"].func.isRequired,
  direction: _propTypes["default"].string,
  isHidden: _propTypes["default"].bool
};
CarouselArrow.defaultProps = {
  isHidden: false
};
var _default = CarouselArrow;
exports["default"] = _default;
