// mc-components/dist/components/Video/Settings.js
"use strict";

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _uniqBy = _interopRequireDefault(require("lodash/uniqBy"));

var _Video = require("./Video");

var _Button = _interopRequireDefault(require("../Button"));

var _Icons = _interopRequireDefault(require("../Icons"));

var _Dropdown = _interopRequireDefault(require("../Dropdown"));

var _DropdownToggle = _interopRequireDefault(require("../DropdownToggle"));

var _DropdownContent = _interopRequireDefault(require("../DropdownContent"));

var _DropdownHeader = _interopRequireDefault(require("../DropdownHeader"));

var _DropdownBody = _interopRequireDefault(require("../DropdownBody"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var Settings = function Settings() {
  var _useContext = (0, _react.useContext)(_Video.VideoContext),
      levels = _useContext.levels,
      level = _useContext.level,
      speed = _useContext.speed,
      setLevel = _useContext.setLevel,
      setSpeed = _useContext.setSpeed;

  var filteredLevels = (0, _uniqBy["default"])(levels, function (l) {
    return l.height;
  });
  return _react["default"].createElement("div", {
    className: "mc-video__settings mc-video__control"
  }, _react["default"].createElement(_Dropdown["default"], {
    placement: "top-end"
  }, _react["default"].createElement(_DropdownToggle["default"], null, _react["default"].createElement(_Button["default"], {
    kind: "link",
    size: "small"
  }, _react["default"].createElement(_Icons["default"], {
    kind: "cog",
    className: "mc-icon mc-icon--4"
  }))), _react["default"].createElement(_DropdownContent["default"], null, _react["default"].createElement(_DropdownHeader["default"], {
    className: "mc-px-4 mc-py-2 mc-text-x-small mc-opacity--muted mc-mr-3"
  }, "Quality"), _react["default"].createElement(_DropdownBody["default"], {
    className: "mc-px-4 mc-py-2"
  }, filteredLevels.map(function (l) {
    return _react["default"].createElement(_Button["default"], {
      key: l.level,
      kind: level === l.level ? 'secondary' : 'link',
      onClick: function onClick() {
        return setLevel(l.level);
      }
    }, l.height, "p");
  })), _react["default"].createElement(_DropdownHeader["default"], {
    className: "mc-px-4 mc-py-2 mc-text-x-small mc-opacity--muted mc-mr-3"
  }, "Speed"), _react["default"].createElement(_DropdownBody["default"], {
    className: "mc-px-4 mc-py-2"
  }, _react["default"].createElement(_Button["default"], {
    kind: speed === 0.5 ? 'secondary' : 'link',
    onClick: function onClick() {
      return setSpeed(0.5);
    }
  }, "0.5x"), _react["default"].createElement(_Button["default"], {
    kind: speed === 1 ? 'secondary' : 'link',
    onClick: function onClick() {
      return setSpeed(1);
    }
  }, "1x"), _react["default"].createElement(_Button["default"], {
    kind: speed === 2 ? 'secondary' : 'link',
    onClick: function onClick() {
      return setSpeed(2);
    }
  }, "2x")))));
};

var _default = Settings;
exports["default"] = _default;
