// mc-components/dist/components/FormGroup/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Error = _interopRequireDefault(require("../Icons/Error"));

var _constants = require("../Forms/constants");

var _utils = require("../Forms/utils");

var _constants2 = require("../constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var FormGroup =
/*#__PURE__*/
function (_PureComponent) {
  _inherits(FormGroup, _PureComponent);

  function FormGroup() {
    _classCallCheck(this, FormGroup);

    return _possibleConstructorReturn(this, _getPrototypeOf(FormGroup).apply(this, arguments));
  }

  _createClass(FormGroup, [{
    key: "render",
    value: function render() {
      var _cn;

      var _this$props = this.props,
          children = _this$props.children,
          className = _this$props.className,
          error = _this$props.error,
          help = _this$props.help,
          label = _this$props.label,
          maxlength = _this$props.maxlength,
          name = _this$props.name,
          optional = _this$props.optional,
          success = _this$props.success,
          touched = _this$props.touched,
          value = _this$props.value;
      var state = (0, _utils.getState)({
        error: error,
        success: success,
        touched: touched
      });
      var classes = (0, _classnames["default"])((_cn = {
        'mc-form-group': true
      }, _defineProperty(_cn, "mc-form-group--".concat(state), state), _defineProperty(_cn, className, className), _cn));
      return _react["default"].createElement("div", {
        className: classes
      }, _react["default"].createElement("div", {
        className: "row no-gutters justify-content-between align-items-center"
      }, _react["default"].createElement("div", {
        className: "col align-self-end"
      }, label && _react["default"].createElement("label", {
        htmlFor: name,
        className: "d-block mc-text-h8 mc-text--left mc-mb-1"
      }, label)), optional && _react["default"].createElement("div", {
        className: "col-auto align-self-end"
      }, _react["default"].createElement("p", {
        className: "mc-text-x-small mc-opacity--silenced mc-text--right mc-mb-1"
      }, "(Optional)")), _react["default"].createElement("div", {
        className: "col-12"
      }, children), _react["default"].createElement("div", {
        className: "col align-self-start"
      }, state === _constants.STATE_DEFAULT && _react["default"].createElement("p", {
        className: "mc-text-x-small mc-opacity--muted mc-text--left mc-mt-1"
      }, help), state === _constants.STATE_ERROR && _react["default"].createElement("p", {
        className: "mc-text-x-small mc-text--error mc-text--left mc-mt-1"
      }, _react["default"].createElement(_Error["default"], null), error), state === _constants.STATE_SUCCESS && _react["default"].createElement("p", {
        className: "mc-text-x-small mc-text--success mc-text--left mc-mt-1"
      }, success)), _react["default"].createElement("div", {
        className: "col-auto align-self-start mc-ml-auto"
      }, maxlength && _react["default"].createElement("p", {
        className: "mc-text-x-small mc-opacity--muted mc-text--right mc-mt-1"
      }, value.length, " / ", maxlength))));
    }
  }]);

  return FormGroup;
}(_react.PureComponent);

exports["default"] = FormGroup;

_defineProperty(FormGroup, "propTypes", {
  children: _constants2.PROP_TYPE_CHILDREN.isRequired,
  className: _propTypes["default"].string,
  error: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].arrayOf(_propTypes["default"].string)]),
  help: _propTypes["default"].string,
  label: _propTypes["default"].string,
  maxlength: _propTypes["default"].number,
  name: _propTypes["default"].string.isRequired,
  optional: _propTypes["default"].bool,
  success: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].arrayOf(_propTypes["default"].string)]),
  touched: _propTypes["default"].bool,
  value: _propTypes["default"].string
});
