// mc-components/dist/components/InputReduxForm/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Input = _interopRequireDefault(require("../Input"));

var _helpers = require("../helpers");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var InputReduxForm = function InputReduxForm(_ref) {
  var input = _ref.input,
      label = _ref.label,
      placeholder = _ref.placeholder,
      _ref$meta = _ref.meta,
      touched = _ref$meta.touched,
      error = _ref$meta.error,
      warning = _ref$meta.warning,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? 'text' : _ref$type;
  var parsedError = (0, _helpers.parseInputErrors)(error);
  var showError = touched && parsedError;
  var showWarning = touched && warning;
  return _react["default"].createElement(_Input["default"], _extends({
    label: label,
    type: type,
    placeholder: placeholder,
    error: showError ? parsedError : '',
    warning: showWarning ? warning : ''
  }, input));
};

InputReduxForm.propTypes = {
  input: _propTypes["default"].object.isRequired,
  label: _propTypes["default"].string,
  type: _propTypes["default"].string,
  meta: _propTypes["default"].object,
  placeholder: _propTypes["default"].string
};
var _default = InputReduxForm;
exports["default"] = _default;
